### PR TITLE
docs(yard): add progress-steps

### PR DESCRIPTION
**Pull request by gergelyke**

### CODE REVIEW

File: documentation-site/components/yard/config/progress-steps.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    current: {
+      value: '0',
+      type: PropTypes.Number,
+      description: 'Defines the current active step index.',
+      stateful: true,
+    },
+    children: {
+      value: `<Step title="Verify Address">
+  <p>Address on file: 1455 Market Street</p>
+  <Button size="compact" onClick={() => setCurrent(1)}>
```
Comment (gergelyke) [inline documentation-site/components/yard/config/progress-steps.ts]: in this case, can I update the `current` prop with this value? @tajo

Comment (tajo) [reply documentation-site/components/yard/config/progress-steps.ts]: sure, this works right? (waiting for the build)

**Comment by tajo** [comment]
Wonder if should have two yard examples for this page (Step vs NumberedStep). 

Also, should we remove overrides / API components?

File: documentation-site/components/yard/config/progress-steps.ts (hunk 1 above)
Comment (tajo) [reply documentation-site/components/yard/config/progress-steps.ts]: Oh I see, it doesn't update the snippet / knob. You could probably try

```jsx
<Button size="compact" onClick={() => {
  setCurrent(1);
  __yard_onChange('ProgressSteps', 'current', 1);
}>
```

It should be taken out (not visible in the code snippet).

Comment (tajo) [reply documentation-site/components/yard/config/progress-steps.ts]: Um, but that would probably work only initially... we could probably enhance the whole `propHook` thing to target arbitrary inner callbacks.

File: documentation-site/components/yard/index.tsx [hunk 2]
```diff
@@ -23,16 +23,18 @@ const YardWrapper: React.FC<
...
+        urlCode={
+          typeof queryStringName !== 'undefined'
+            ? router.query[queryStringName]
+            : router.query.code
+        }
+        queryStringName={queryStringName}
+        placeholderElement={() => (
           if (!placeholderHeight) return null;
-
-          return (
```
Comment (tajo) [inline documentation-site/components/yard/index.tsx]: does this `return`  disappeared?

Comment (gergelyke) [reply documentation-site/components/yard/index.tsx]: just fixed it - i didn't resolve this merge conflict properly 🤦‍♂

**Comment by tajo** [comment]
Added a new feature - `propHook` now also accepts a function where you can define your own visitor to locate an arbitrary function that should be instrumented.

It solves the use-case of this component where `onClick` is deeply nested in a children prop.

This is a very flexible API since you can arbitrarily traverse the tree.

**Comment by gergelyke** [comment]
> Added a new feature - `propHook` now also accepts a function where you can define your own visitor to locate an arbitrary function that should be instrumented.
> 
> It solves the use-case of this component where `onClick` is deeply nested in a children prop.
> 
> This is a very flexible API since you can arbitrarily traverse the tree. It can cover `renderProps` pattern as well.

this is very cool, thanks Vojtech!

**Pull request merged**